daniel-d-truong/tag.it
# Implemented the Flow For Adding User to a Course

How to Test:
  - Log In
  - Go to localhost:3000/add
  - Type in CSE 100 / CSE 101 in the textboxes
  - Check in Firebase that your user id is being added to the course objects and that in your user object, the course id is being added to the `studentCourseList` array
  - If you want to test again, you'll have to manually delete these from Firebase and try again

Notes:
  - Should work when adding 2 or more classes as well (purpose of the synchronous reduce line)
  - Added checks in backend to ensure that the same course could not be added twice to a user and that a user's courseList would not have duplicates
  - Dealt with routing for the home page so (@JIzabal) shouldn't need to worry about that now
  - Currently waiting on the COURSES routes in order to retrieve all of the course objects 

Resources: 
  - [Explanation of Reduce & Synchronous Promises](https://css-tricks.com/why-using-reduce-to-sequentially-resolve-promises-works/)
